Connection mvp

zacharymarshal opened the pull request.

zacharymarshal committed:
New Connection object

Allows you to run simple queries to start. Need more tests and edge
cases

zacharymarshal committed:
Update tests for querying postgers

zacharymarshal committed:
Fix querying empty values

zacharymarshal committed:
Fix issue with the default socket timeout

zacharymarshal committed:
Update phpunit, clean up tests

lightster commented on src/Connection.php:
```diff
@@ -0,0 +1,176 @@
+    /**
+     * @param $url string postgres://user:pass@localhost:5432/dbname?param=val
+     */
+    public function __construct(string $url, int $connect_timeout = null)
```
Have you considered Java for this?

lightster commented on src/Connection.php:
```diff
@@ -0,0 +1,176 @@
+                $num_cols = $buf->readInt16();
+                for ($i = 0; $i < $num_cols; $i++) {
+                    $col_val_len = $buf->readInt32();
+                    if ($col_val_len === 4294967295) {
```
You might consider `0xFFFFFFFF` here instead of `4294967295`.

zacharyrankin replied:
lol yea, is clearer

zacharymarshal committed:
Use 0xFFFFFFFF for checking col values of -1

https://stackoverflow.com/questions/31246843/why-does-4294967295-the-highest-number-at-32bit-equal-1

zacharymarshal committed:
Update composer test command

zacharymarshal committed:
Update to PHP 7.1

zacharymarshal committed:
Authenticate using plaintext passwords

zacharymarshal committed:
Cleanup some variable names and comments

zacharymarshal committed:
Implement md5 password support

zacharymarshal committed:
Improve performance of ReadBuffers

Instead of making a copy of the string, keep track of the offset and use
that.

zacharymarshal committed:
Work in progress with Errors

Didn't want to lose the work so just keeping what I have for now

zacharymarshal committed:
Update docs for hba conf

zacharymarshal committed:
Update connection error handling

zacharymarshal committed:
Update travis for postgres
